cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/harbor_api_v2_repo_lister.go

```diff
@@ -0,0 +1,298 @@
+	log.Infof("Harbor v2 Registry [%s] Ping: %s", ociRepo.url.String(), pong)
+	if err != nil {
+		return false, err
+	}
+
+	projectName, err := harborProjectNameFromURL(ociRepo.url)
+	if err != nil {
+		return false, err
+	}
+
+	err = harborProjectExists(ctx, projectName, parsedRef, cred)
+	return err == nil, err
+}
+
+func (l *harborRegistryApiV2RepositoryLister) ListRepositoryNames(ociRepo *OCIChartRepository) ([]string, error) {
+	log.Infof("+ListRepositoryNames(%s)", ociRepo.url.String())
+	ref := strings.TrimPrefix(ociRepo.url.String(), fmt.Sprintf("%s://", registry.OCIScheme))
+	parsedRef, err := orasregistryv2.ParseReference(ref)
+	if err != nil {
+		return nil, err
+	}
+
+	projectName, err := harborProjectNameFromURL(ociRepo.url)
+	if err != nil {
+		return nil, err
+	}
+	repos := []string{}
+	ctx := context.Background()
+	cred, err := ociRepo.registryCredentialFn(ctx, parsedRef.Host())
+	if err != nil {
+		return nil, err
+	}
+	for page, pageSize, more := 1, 10, true; more; page++ {
+		onePage, err := listHarborProjectRepositories(
+			ctx, projectName, parsedRef, cred, page, pageSize)
+		if err != nil {
+			return nil, err
+		}
+		repos = append(repos, onePage...)
+		if len(onePage) < pageSize {
+			more = false
+		}
+	}
+	return repos, nil
+}
+
+// ref https://demo.goharbor.io/#/ping/getPing
+func pingHarbor(ctx context.Context, ref orasregistryv2.Reference, cred orasregistryauthv2.Credential) (string, error) {
+	log.Infof("+pingHarbor")
+	url := fmt.Sprintf("%s/ping", buildHarborRegistryBaseURL(false, ref))
+	resp, err := harborDoHttpRequest(ctx, http.MethodGet, cred, url)
+	if err != nil {
+		return "", err
+	}
+	contentType := resp.Header.Get("Content-Type")
+	if !strings.HasPrefix(contentType, "text/plain") {
+		return "", fmt.Errorf("unexpected response content type: [%s]", contentType)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		lr := io.LimitReader(resp.Body, 100)
+		if pong, err := ioutil.ReadAll(lr); err != nil {
+			return "", err
+		} else if string(pong) != "Pong" {
+			return "", fmt.Errorf("unexpected response body: %s", string(pong))
+		} else {
+			return string(pong), nil
+		}
+
+	case http.StatusNotFound:
+		return "", errdef.ErrNotFound
+
+	default:
+		err = parseHarborErrorResponse(resp)
+		return err.Error(), err
+	}
+}
+
+//  https://demo.goharbor.io/#/project/headProject
+func harborProjectExists(ctx context.Context, projectName string, ref orasregistryv2.Reference, cred orasregistryauthv2.Credential) error {
+	log.Infof("+projectExists(%s)", projectName)
+	url := fmt.Sprintf("%s/projects?project_name=%s",
+		buildHarborRegistryBaseURL(false, ref), projectName)
+	resp, err := harborDoHttpRequest(ctx, http.MethodHead, cred, url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return nil
+
+	case http.StatusNotFound:
+		return errdef.ErrNotFound
+
+	case http.StatusForbidden:
+		return errors.New("forbidden")
```

Comment:
I don't know where both errors are being handled, but it seems we are mixing up the error builder from oras (`errdef`) and the builtin `errors` packages. Should we use the same for both cases? Or perhaps it doesn't really matter due to the error hanlding being performed on the call site?

Reply:
I will get rid of lines 157-158. They are not needed.